sematic/types/registry.py

```diff
@@ -244,7 +245,16 @@ def _get_registered_func(
     validate_type_annotation(type_)
     registry_type = get_origin_type(type_)
 
-    return registry.get(registry_type)
+    registered = registry.get(registry_type)
+    try:
+        is_enum = issubclass(type_, Enum)
+    except TypeError:
+        is_enum = False
+    if registered is None and is_enum:
+        # enum types can register their own handlers, but if they don't
+        # we can use the default enum handler
+        registered = registry.get(Enum)
+    return registered
```

Comment:
I think we have a similar mechanism with dataclasses. If they have they own handlers, we use it, otherwise, we use the default dataclass one. See `DataclassKey`:
https://github.com/sematic-ai/sematic/blob/9587369325ac13c181ae0bf8c1bfac937c0d250f/sematic/types/casting.py#L46-L48

Maybe we can use a similar mechanism here in `casting.py` instead of here.

Reply:
Made consistent for now and filed #193